internal/cmd/io_binary.go

```diff
@@ -76,17 +75,31 @@ func NewBinaryIO(ctx context.Context, id string, uri *url.URL) (_ UpstreamIO, er
 		return nil, err
 	}
 
-	b := make([]byte, 1)
-	if _, err := w.Read(b); err != nil && err != io.EOF {
```

Comment:
this would not cause an error, even if the `wait` pipe doesn't get closed. Neither would `cmd.Start()` result in an issue, when we have the binary exit